Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -1,47 +1,150 @@
-                new[] { "tFc", "tF", "Ftc" }, 
+                new[] { 3, 4, 1, 5 },
+                new[] { 2, 8, 5, 1 },
+                new[] { 5, 2, 4, 4 },
+                new[] { "tFc", "tF", "Ftc" },
                 new[] { 3, 2, 0 });
             Test(
-                new[] { 18, 86, 76, 0, 34, 30, 95, 12, 21 }, 
-                new[] { 26, 56, 3, 45, 88, 0, 10, 27, 53 }, 
-                new[] { 93, 96, 13, 95, 98, 18, 59, 49, 86 }, 
-                new[] { "f", "Pt", "PT", "fT", "Cp", "C", "t", "", "cCp", "ttp", "PCFt", "P", "pCt", "cP", "Pc" }, 
+                new[] { 18, 86, 76, 0, 34, 30, 95, 12, 21 },
+                new[] { 26, 56, 3, 45, 88, 0, 10, 27, 53 },
+                new[] { 93, 96, 13, 95, 98, 18, 59, 49, 86 },
+                new[] { "f", "Pt", "PT", "fT", "Cp", "C", "t", "", "cCp", "ttp", "PCFt", "P", "pCt", "cP", "Pc" },
                 new[] { 2, 6, 6, 2, 4, 4, 5, 0, 5, 5, 6, 6, 3, 5, 6 });
             Console.ReadKey(true);
         }
 
-        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
-        {
+        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
+        {
             var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
             Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
             Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
-            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
             Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
-            Console.WriteLine(result);
+            Console.WriteLine(result);
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
+            int[] output = new int[dietPlans.Length];
+            int[] calories = new int[protein.Length];
+
+
+            for (int i = 0; i < protein.Length; i++)
+                calories[i] = protein[i] * 5 + carbs[i] * 5 + fat[i] * 9;
+
+
+            for (int i = 0; i < dietPlans.Length; i++)
+            {
+                List<int> temp = new List<int>();
+                List<int> List = new List<int>();
+                int min = 10000, max = -1;
+                for (int j = 0; j < protein.Length; j++)
+                    List.Add(j);
+                for (int j = 0; j < dietPlans[i].Length; j++)
+                {
+                    temp = new List<int>();
+                    string tempDietPlan = dietPlans[i];
+                    if (tempDietPlan[j] == 'C')
+                    {
+                        foreach (int k in List)

```

Comment:
I recommend you to read Linq. You can achieve many inbuilt functionality in this
like - Max, Min, Sum for a list

Reply:
We cannot use the Linq functions here as in List we are storing only selected meals' indices and not all the indices.